src/providers/oracle/qgsoracleprovider.cpp

```diff
@@ -608,9 +610,16 @@ bool QgsOracleProvider::loadFields()
                  ",t.data_scale"
                  ",t.char_length"
                  ",t.char_used"
-                 ",t.data_default"
-                 " FROM all_tab_columns t"
-                 " WHERE t.owner=? AND t.table_name=?" ) ;
+                 ",t.data_default" +
+                 QString( mOracleVersion >= 12 ?
+                          ",CASE WHEN t.virtual_column = 'YES' OR a.generation_type = 'ALWAYS' THEN 'YES' ELSE 'NO' END" :
+                          ",t.virtual_column" ) +
+                 " FROM all_tab_cols t" +
+                 QString( mOracleVersion >= 12 ?
+                          " LEFT JOIN all_tab_identity_cols a ON a.column_name = t.column_name AND a.owner = t.owner AND a.table_name = t.table_name" :
+                          "" ) +
```

Comment:
Identity column needs 12th version of Oracle but virtual column are available since the 11th version, isn't it ? I don't know if it's will concern many users.

Reply:
If *mOracleVersion* is lower than 12, then I still get the *t.virtual_column* (which is already set with "YES" if it's a virtual column), so I'd say it should work with version 11.

Reply:
Indeed, sorry I misread the line between the condition in SQL inside the condition in C++...